core/src/main/java/org/bitcoinj/core/PeerAddress.java

```diff
@@ -331,7 +331,7 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         PeerAddress other = (PeerAddress) o;
-        return other.addr.equals(addr) && other.port == port && other.time == time && other.services.equals(services);
```

Comment:
Should `time` be removed from `hashCode` too?

Reply:
Good catch!

Reply:
Yes, I think it should be removed from `hashCode()` too.

Reply:
I force-pushed an update with `time` removed from `hashCode()` as well.